src/app/[slug]/page.jsx

```diff
@@ -1,43 +1,27 @@
 import PageHead from "@/components/PageHead";
-import { AllPages, SinglePage } from "@/queries/pages";
+import { SinglePage } from "@/queries/pages";
 import { RichText } from "@graphcms/rich-text-react-renderer";
 import Head from "next/head";
 
-export async function getStaticPaths() {
-  const allPages = await fetch(process.env.HYGRAPH_ENDPOINT, {
-    method: "POST",
-    headers: {
-      "Content-Type": "application/json",
-    },
-    body: JSON.stringify({
-      query: AllPages,
-    }),
-  })
-    .then((res) => res.json())
-    .then((res) => res.data.pages);
-  const paths = allPages.map((page) => ({
-    params: { slug: page.slug },
-  }));
-  return { paths, fallback: false };
-}
 
-export async function getStaticProps({ params }) {
-  const { page } = await fetch(process.env.HYGRAPH_ENDPOINT, {
+async function getPage(slug) {
+    const { page } = await fetch(process.env.HYGRAPH_ENDPOINT, {
     method: "POST",
```

Comment:
It's better to use "GET" request when fetching data, to utilize automatic request deduplication Next.js provides. Explained [here](https://nextjs.org/docs/app/building-your-application/data-fetching/caching#graphql-and-cache). Although you use the fetch function only once here, you'll probably use it again when generating metadata for SEO.

Reply:
As discussed in Slack, it seems the latest on the App router lets POST get deduped as long as they're not in POST API routes